// traffic-violation-frontend/src/context/ToastContext.jsx
import { createContext, useContext, useState, useCallback } from 'react'

const ToastContext = createContext()

export function useToast() {
  const context = useContext(ToastContext)
  if (!context) {
    throw new Error('useToast must be used within ToastProvider')
  }
  return context
}

export function ToastProvider({ children }) {
  const [toasts, setToasts] = useState([])

  const addToast = useCallback((message, type = 'success', duration = 3000) => {
    const id = Date.now()
    setToasts(prev => [...prev, { id, message, type }])
    
    setTimeout(() => {
      setToasts(prev => prev.filter(toast => toast.id !== id))
    }, duration)
  }, [])

  const removeToast = useCallback((id) => {
    setToasts(prev => prev.filter(toast => toast.id !== id))
  }, [])

  const success = (message) => addToast(message, 'success')
  const error = (message) => addToast(message, 'error')
  const warning = (message) => addToast(message, 'warning')
  const info = (message) => addToast(message, 'info')

  return (
    <ToastContext.Provider value={{ success, error, warning, info }}>
      {children}
      <ToastContainer toasts={toasts} onRemove={removeToast} />
    </ToastContext.Provider>
  )
}

function ToastContainer({ toasts, onRemove }) {
  if (toasts.length === 0) return null

  const getToastColors = (type) => {
    switch (type) {
      case 'success':
        return { bg: 'rgba(6, 95, 70, 0.95)', border: '#047857', text: '#ffffff' }
      case 'error':
        return { bg: 'rgba(153, 27, 27, 0.95)', border: '#7f1d1d', text: '#ffffff' }
      case 'warning':
        return { bg: 'rgba(146, 64, 14, 0.95)', border: '#78350f', text: '#ffffff' }
      case 'info':
        return { bg: 'rgba(30, 64, 175, 0.95)', border: '#1e3a8a', text: '#ffffff' }
      default:
        return { bg: 'rgba(31, 41, 55, 0.95)', border: '#111827', text: '#ffffff' }
    }
  }

  const getIcon = (type) => {
    switch (type) {
      case 'success':
        return '✅'
      case 'error':
        return '❌'
      case 'warning':
        return '⚠️'
      case 'info':
        return 'ℹ️'
      default:
        return '📌'
    }
  }

  return (
    <div style={{ position: 'fixed', top: '20px', right: '20px', zIndex: 99999, display: 'flex', flexDirection: 'column', gap: '12px', maxWidth: '400px', width: '90vw' }}>
      {toasts.map((toast) => {
        const colors = getToastColors(toast.type)
        return (
          <div
            key={toast.id}
            style={{
              display: 'flex', alignItems: 'center', gap: '14px',
              padding: '16px 20px',
              backgroundColor: colors.bg,
              borderLeft: `6px solid ${colors.border}`,
              color: colors.text,
              borderRadius: '8px',
              boxShadow: '0 10px 25px -5px rgba(0, 0, 0, 0.3)',
              backdropFilter: 'blur(8px)',
              animation: 'slideIn 0.3s ease-out'
            }}
          >
            <span style={{ fontSize: '22px' }}>{getIcon(toast.type)}</span>
            <p style={{ margin: 0, fontSize: '15px', fontWeight: 700, lineHeight: 1.4, flex: 1, textShadow: '0 1px 2px rgba(0,0,0,0.5)' }}>
              {toast.message}
            </p>
            <button
              onClick={() => onRemove(toast.id)}
              style={{
                background: 'none', border: 'none', color: '#fff',
                fontSize: '18px', cursor: 'pointer', padding: '4px',
                opacity: 0.7, alignSelf: 'flex-start'
              }}
              onMouseEnter={e => e.currentTarget.style.opacity = 1}
              onMouseLeave={e => e.currentTarget.style.opacity = 0.7}
            >
              ✖
            </button>
          </div>
        )
      })}
    </div>
  )
}
