// traffic-violation-frontend/src/pages/PoliceRegister.jsx
import { API_BASE_URL } from '../config';
import { useState } from 'react'
import { Link, useNavigate } from 'react-router-dom'
import { useToast } from '../context/ToastContext'

function PoliceRegister() {
  const navigate = useNavigate()
  const { success, error: showError } = useToast()
  const [formData, setFormData] = useState({
    full_name: '', email: '', phone_no: '', password: '', confirm_password: '', district: 'Sivagangai'
  })
  const [loading, setLoading] = useState(false)
  const [errors, setErrors] = useState({})
  const [showPw, setShowPw] = useState(false)
  const [showCpw, setShowCpw] = useState(false)

  const validate = () => {
    const e = {}
    if (!formData.full_name.trim()) e.full_name = 'Required'
    if (!formData.email.trim() || !/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(formData.email)) e.email = 'Valid email required'
    if (!formData.phone_no || !/^[0-9]{10}$/.test(formData.phone_no.trim())) e.phone_no = '10-digit number required'
    if (!formData.password || formData.password.length < 8) e.password = 'Min 8 characters'
    if (formData.password !== formData.confirm_password) e.confirm_password = 'Passwords do not match'
    setErrors(e)
    return Object.keys(e).length === 0
  }

  const handleChange = (e) => {
    const { name, value } = e.target
    setFormData(prev => ({ ...prev, [name]: value }))
    if (errors[name]) setErrors(prev => ({ ...prev, [name]: '' }))
  }

  const handleSubmit = async (e) => {
    e.preventDefault()
    if (!validate()) return
    setLoading(true)
    try {
      const res = await fetch(`${API_BASE_URL}/api/auth/police/register`, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          full_name: formData.full_name.trim(),
          email: formData.email.trim().toLowerCase(),
          phone_no: formData.phone_no,
          password: formData.password,
          confirm_password: formData.confirm_password,
          district: formData.district
        })
      })
      const data = await res.json()
      if (!res.ok) {
        let message = 'Registration failed'
        if (typeof data?.detail === 'string') message = data.detail
        else if (Array.isArray(data?.detail)) message = data.detail.map(e => e.msg || JSON.stringify(e)).join(', ')
        throw new Error(message)
      }
      success('Officer account created!')
      setTimeout(() => navigate('/police/login'), 1500)
    } catch (err) {
      showError(err.message)
    } finally {
      setLoading(false)
    }
  }

  const inputStyle = {
    width: '100%', boxSizing: 'border-box',
    padding: '13px 16px',
    background: '#fafaf9',
    border: '1.5px solid #d6d3d1',
    borderRadius: '8px',
    color: '#111827', fontSize: '14px', outline: 'none',
    transition: 'all 0.18s', fontFamily: 'inherit',
  }
  const onFocus = (e) => {
    e.target.style.borderColor = '#78350f'
    e.target.style.background = '#fff'
    e.target.style.boxShadow = '0 0 0 3px rgba(120,53,15,0.10)'
  }
  const onBlur = (e) => {
    e.target.style.borderColor = '#d6d3d1'
    e.target.style.background = '#fafaf9'
    e.target.style.boxShadow = 'none'
  }
  const lbl = { display: 'block', color: '#374151', fontSize: '12px', fontWeight: 700, marginBottom: '6px', letterSpacing: '0.2px' }
  const errTxt = { color: '#dc2626', fontSize: '11px', marginTop: '4px', margin: '4px 0 0' }

  return (
    <div style={{ minHeight: '100vh', width: '100%', display: 'flex' }}>

      {}
      <div style={{
        flex: '1 1 55%',
        position: 'relative',
        backgroundImage: 'url(/police_login_bg.png)',
        backgroundSize: 'cover',
        backgroundPosition: 'center top',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'flex-end',
        padding: '48px',
        minHeight: '100vh',
      }}>
        <div style={{ position: 'absolute', inset: 0, background: 'linear-gradient(to bottom, rgba(15,23,42,0.20) 0%, rgba(15,23,42,0.80) 100%)' }} />

        {}
        <div style={{ position: 'absolute', top: '32px', left: '48px', zIndex: 2, display: 'flex', alignItems: 'center', gap: '12px' }}>
          <div style={{ width: '40px', height: '40px', background: 'rgba(255,255,255,0.14)', borderRadius: '10px', border: '1.5px solid rgba(255,255,255,0.30)', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
            <svg width="20" height="20" viewBox="0 0 24 24" fill="none">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="1.8" d="M9 12l2 2 4-4M20.618 5.984A11.955 11.955 0 0112 2.944a11.955 11.955 0 01-8.618 3.04A12.02 12.02 0 003 9c0 5.591 3.824 10.29 9 11.622 5.176-1.332 9-6.03 9-11.622 0-1.042-.133-2.052-.382-3.016z" stroke="white"/>
            </svg>
          </div>
          <div>
            <p style={{ color: '#fff', fontWeight: 800, fontSize: '16px', margin: 0, lineHeight: 1.2 }}>Marga Rakshak</p>
            <p style={{ color: 'rgba(255,255,255,0.6)', fontSize: '10px', margin: 0, letterSpacing: '1px', textTransform: 'uppercase' }}>Tamil Nadu Police</p>
          </div>
        </div>

        {}
        <div style={{ position: 'absolute', top: '36px', right: '48px', zIndex: 2 }}>
          <div style={{ display: 'inline-flex', alignItems: 'center', gap: '6px', background: 'rgba(251,191,36,0.15)', border: '1px solid rgba(251,191,36,0.40)', borderRadius: '999px', padding: '5px 14px' }}>
            <svg width="10" height="10" fill="#fbbf24" viewBox="0 0 24 24"><path d="M12 22s8-4 8-10V5l-8-3-8 3v7c0 6 8 10 8 10z"/></svg>
            <span style={{ color: '#fbbf24', fontSize: '10px', fontWeight: 700, letterSpacing: '0.8px', textTransform: 'uppercase' }}>Restricted Access</span>
          </div>
        </div>

        {}
        <div style={{ position: 'relative', zIndex: 2 }}>
          <div style={{ display: 'inline-flex', alignItems: 'center', gap: '6px', background: 'rgba(251,191,36,0.12)', border: '1px solid rgba(251,191,36,0.30)', borderRadius: '6px', padding: '5px 12px', marginBottom: '20px' }}>
            <svg width="12" height="12" fill="none" stroke="#fbbf24" viewBox="0 0 24 24">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M12 9v4M12 17h.01M10.29 3.86L1.82 18a2 2 0 001.71 3h16.94a2 2 0 001.71-3L13.71 3.86a2 2 0 00-3.42 0z"/>
            </svg>
            <span style={{ color: '#fbbf24', fontSize: '11px', fontWeight: 600 }}>Authorised Personnel Only</span>
          </div>

          <h2 style={{ color: '#fff', fontSize: 'clamp(28px, 3.5vw, 44px)', fontWeight: 900, margin: '0 0 12px', lineHeight: 1.1, letterSpacing: '-0.5px' }}>
            Officer<br />Registration
          </h2>
          <p style={{ color: 'rgba(255,255,255,0.68)', fontSize: '15px', margin: '0 0 28px', lineHeight: 1.6, maxWidth: '380px' }}>
            Create your official Tamil Nadu Traffic Police enforcement account. All registrations are verified by the department before activation.
          </p>

          <div style={{ display: 'flex', gap: '28px', flexWrap: 'wrap' }}>
            {[{ value: 'Verified', label: 'Admin Approval Required' }, { value: 'SSL', label: 'Encrypted Registration' }, { value: '24h', label: 'Activation SLA' }].map((s, i) => (
              <div key={i}>
                <p style={{ color: '#fff', fontWeight: 800, fontSize: '18px', margin: 0 }}>{s.value}</p>
                <p style={{ color: 'rgba(255,255,255,0.45)', fontSize: '11px', margin: 0 }}>{s.label}</p>
              </div>
            ))}
          </div>
        </div>
      </div>

      {}
      <div style={{
        flex: '0 0 440px',
        background: '#fff',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        padding: '48px 44px',
        boxShadow: '-8px 0 40px rgba(0,0,0,0.12)',
        position: 'relative',
        zIndex: 2,
        overflowY: 'auto',
      }}>

        {}
        <div style={{ position: 'absolute', top: '20px', right: '20px' }}>
          <Link to="/police/login" style={{ display: 'inline-flex', alignItems: 'center', gap: '6px', background: '#f1f5f9', border: '1px solid #e2e8f0', borderRadius: '8px', padding: '7px 14px', color: '#334155', fontSize: '12px', fontWeight: 600, textDecoration: 'none' }}>
            ← Officer Sign In
          </Link>
        </div>

        {}
        <div style={{ marginBottom: '28px' }}>
          {}
          <div className="gov-portal-text-amber fade-up-enter" style={{ marginBottom: '12px' }}>
            <svg width="10" height="10" fill="currentColor" viewBox="0 0 24 24">
              <path d="M12 22s8-4 8-10V5l-8-3-8 3v7c0 6 8 10 8 10z"/>
            </svg>
            Government of Tamil Nadu — Traffic Enforcement Portal
          </div>
          <div style={{ display: 'inline-block', padding: '4px 12px', background: '#fef3c7', borderRadius: '6px', color: '#92400e', fontSize: '11px', fontWeight: 700, letterSpacing: '0.8px', textTransform: 'uppercase', marginBottom: '14px' }}>
            Officer Registration
          </div>
          <h1 style={{ color: '#451a03', fontSize: '24px', fontWeight: 800, margin: '0 0 6px', lineHeight: 1.2 }}>
            Create Officer Account
          </h1>
          <p style={{ color: '#78716c', fontSize: '13px', margin: 0 }}>
            Official law enforcement registration — Tamil Nadu Police
          </p>
        </div>

        {}
        <div style={{ background: '#fef3c7', border: '1px solid #fcd34d', borderRadius: '8px', padding: '10px 14px', display: 'flex', alignItems: 'center', gap: '8px', marginBottom: '24px' }}>
          <svg width="14" height="14" viewBox="0 0 24 24" fill="none" stroke="#92400e" strokeWidth="2">
            <path d="M12 9v4M12 17h.01M10.29 3.86L1.82 18a2 2 0 001.71 3h16.94a2 2 0 001.71-3L13.71 3.86a2 2 0 00-3.42 0z"/>
          </svg>
          <span style={{ color: '#92400e', fontSize: '12px', fontWeight: 600 }}>Restricted — Authorised Law Enforcement Only</span>
        </div>

        <form onSubmit={handleSubmit} style={{ display: 'flex', flexDirection: 'column', gap: '16px' }}>
          <div>
            <label style={lbl}>Full Name</label>
            <input type="text" name="full_name" value={formData.full_name} onChange={handleChange}
              placeholder="Officer full name" required style={inputStyle} onFocus={onFocus} onBlur={onBlur}/>
            {errors.full_name && <p style={errTxt}>{errors.full_name}</p>}
          </div>

          <div>
            <label style={lbl}>Official Email Address</label>
            <input type="email" name="email" value={formData.email} onChange={handleChange}
              placeholder="officer@tnpolice.gov.in" required style={inputStyle} onFocus={onFocus} onBlur={onBlur}/>
            {errors.email && <p style={errTxt}>{errors.email}</p>}
          </div>

          <div style={{ display: 'grid', gridTemplateColumns: '1fr 1fr', gap: '12px' }}>
            <div>
              <label style={lbl}>Phone Number</label>
              <input type="tel" name="phone_no" value={formData.phone_no} onChange={handleChange}
                placeholder="10-digit mobile number" maxLength={10} required style={inputStyle} onFocus={onFocus} onBlur={onBlur}/>
              {errors.phone_no && <p style={errTxt}>{errors.phone_no}</p>}
            </div>
            <div>
              <label style={lbl}>District</label>
              <select name="district" value={formData.district} onChange={handleChange} required
                style={{ ...inputStyle, cursor: 'pointer' }} onFocus={onFocus} onBlur={onBlur}>
                {['Chennai', 'Coimbatore', 'Madurai', 'Trichy', 'Salem', 'Tirunelveli', 'Sivagangai', 'Other'].map(v => (
                  <option key={v} value={v}>{v}</option>
                ))}
              </select>
            </div>
          </div>

          <div style={{ display: 'grid', gridTemplateColumns: '1fr 1fr', gap: '12px' }}>
            <div>
              <label style={lbl}>Password</label>
              <div style={{ position: 'relative' }}>
                <input type={showPw ? 'text' : 'password'} name="password" value={formData.password} onChange={handleChange}
                  placeholder="Min 8 chars" required style={{ ...inputStyle, paddingRight: '40px' }} onFocus={onFocus} onBlur={onBlur}/>
                <button type="button" onClick={() => setShowPw(!showPw)} tabIndex={-1}
                  style={{ position: 'absolute', right: '12px', top: '50%', transform: 'translateY(-50%)', background: 'none', border: 'none', color: '#a8a29e', cursor: 'pointer', padding: 0 }}>
                  <svg width="15" height="15" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2"><path d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"/><circle cx="12" cy="12" r="3"/></svg>
                </button>
              </div>
              {errors.password && <p style={errTxt}>{errors.password}</p>}
            </div>
            <div>
              <label style={lbl}>Confirm Password</label>
              <div style={{ position: 'relative' }}>
                <input type={showCpw ? 'text' : 'password'} name="confirm_password" value={formData.confirm_password} onChange={handleChange}
                  placeholder="Re-enter" required style={{ ...inputStyle, paddingRight: '40px' }} onFocus={onFocus} onBlur={onBlur}/>
                <button type="button" onClick={() => setShowCpw(!showCpw)} tabIndex={-1}
                  style={{ position: 'absolute', right: '12px', top: '50%', transform: 'translateY(-50%)', background: 'none', border: 'none', color: '#a8a29e', cursor: 'pointer', padding: 0 }}>
                  <svg width="15" height="15" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2"><path d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z"/><circle cx="12" cy="12" r="3"/></svg>
                </button>
              </div>
              {errors.confirm_password && <p style={errTxt}>{errors.confirm_password}</p>}
            </div>
          </div>

          {formData.password && formData.password.length >= 8 && formData.password === formData.confirm_password && (
            <div style={{ padding: '8px 12px', background: '#f0fdf4', border: '1px solid #bbf7d0', borderRadius: '8px' }}>
              <p style={{ color: '#16a34a', fontSize: '12px', margin: 0, fontWeight: 600 }}>✓ Passwords match</p>
            </div>
          )}

          <button type="submit" disabled={loading} style={{
            width: '100%', padding: '15px',
            background: loading ? '#a8a29e' : 'linear-gradient(135deg, #78350f 0%, #451a03 100%)',
            color: '#fff', fontWeight: 800, fontSize: '15px',
            border: 'none', borderRadius: '8px',
            cursor: loading ? 'not-allowed' : 'pointer',
            boxShadow: loading ? 'none' : '0 4px 16px rgba(69,26,3,0.38)',
            transition: 'all 0.18s', marginTop: '4px',
          }}>
            {loading ? 'Creating Account…' : 'Create Officer Account →'}
          </button>
        </form>

        <div style={{ display: 'flex', alignItems: 'center', gap: '12px', margin: '22px 0 18px' }}>
          <div style={{ flex: 1, height: '1px', background: '#e7e5e4' }} />
          <span style={{ color: '#a8a29e', fontSize: '12px' }}>or</span>
          <div style={{ flex: 1, height: '1px', background: '#e7e5e4' }} />
        </div>

        <div style={{ display: 'flex', flexDirection: 'column', gap: '9px', textAlign: 'center' }}>
          <p style={{ color: '#78716c', fontSize: '13px', margin: 0 }}>
            Already have an account?{' '}
            <Link to="/police/login" style={{ color: '#78350f', fontWeight: 700, textDecoration: 'none' }}>Officer Sign In</Link>
          </p>
          <p style={{ color: '#78716c', fontSize: '13px', margin: 0 }}>
            Not an officer?{' '}
            <Link to="/register" style={{ color: '#78350f', fontWeight: 700, textDecoration: 'none' }}>Citizen Registration</Link>
          </p>
        </div>

        <p style={{ color: '#cbd5e1', fontSize: '11px', textAlign: 'center', marginTop: '28px' }}>
          © 2026 Marga Rakshak · Tamil Nadu Police · Authorized Access Only
        </p>
      </div>
    </div>
  )
}

export default PoliceRegister
